Modules/_sqlite/cursor.c

```diff
@@ -925,6 +931,11 @@ _pysqlite_query_execute(pysqlite_Cursor* self, int multiple, PyObject* operation
         if (!parameters) {
             break;
         }
+        // PyIter_Next() may have a side-effect on the connection's state.
+        // See: https://github.com/python/cpython/issues/143198.
+        if (!pysqlite_check_connection(self->connection)) {
```

Comment:
```suggestion
        if (multiple && !pysqlite_check_connection(self->connection)) {
```

Strictly speaking, this check is redundant when dealing with `mutliple = 0` because `parameters_iter` is a true `list_iterator` that is safely constructed and `parameters` may be an evil one. With `multiple = 1`, we need to check this at every iteration as `parameters_iter` may come from user-defined code.